Automattic/jetpack
# Podcast Player (previous "Episodes") Block Edit

This PR provides the basis for Podcast episodes Block's edit definition. 

It builds on https://github.com/Automattic/jetpack/pull/14928

#### Changes proposed in this Pull Request:
* Rename to `Podcast Player` (previously `Podcast Episodes`)
* Provide PHP code required to parse feed for audio.
* Supply audio data to Block edit/save.
* Scaffold out basic Block render.


#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
`paYJgx-sH-p2`

#### Testing instructions:

Testing RSS feed: https://anchor.fm/s/9400d7c/podcast/rss

* Insert Block
* Type invalid URL into input box. See warning.
* Add valid AnchorFM RSS feed into box (see above) - submit.
* See "Loading" and then blank Block.
* Inspect Block output - see `<script>` tag container track data and Block HTML "placeholder" (similar to the below):
<img width="555" alt="Screenshot 2020-03-13 at 09 53 35" src="https://user-images.githubusercontent.com/444434/76610332-aa580a00-6510-11ea-8010-7491db574a50.png">
* Check no console errors.

#### Proposed changelog entry for your changes:

*

## Todo List
- [x] Rename Block to `Podcast Player`.
- [x] Rebase from `master` so we are up to date with merged scaffold.
- [x] Check for `url` attr in `render_block` and if not available then return suitable response.
- [x] Minor: convert `forEach` in `get_track_list` to `array_map`.
- [x] Handle checking for `WP_Error` responses from `get_track_list` call in `render_block`.
- [x] Add checks for presence of `url` and `type` data within each `$episode` returned from RSS. We can't guarantee they are available and if not then they should be dropped from the tracklist.
- [x] Rename `$data` to designate it's purpose inside `render_block`.
- [x] Fix CI build errors
- [x] ~Fix synced WPCom commit.~ Not required (see [here](https://github.com/Automattic/jetpack/pull/14948#issuecomment-598212895))


## Followup PR tasks
- Refactor track list data to conform to https://schema.org/AudioObject.
- Add regex to test for valid AnchorFM embed URLs ([example](https://github.com/Automattic/jetpack/blob/f02d252966e432088b4c3bcc76c95729b5a885a6/extensions/blocks/pinterest/edit.js#L17)).

Fixes #14939.
